buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -2807,9 +2810,17 @@ class BeamModulePlugin implements Plugin<Project> {
         throw new GradleException(exceptionMessage)
       }
 
+      // Transform service delivers transforms that refer to SDK harness containers with following sufixes.
+      def transformServiceJavaContainerSuffix = 'java11'
```

Comment:
what caused this test requiring two java/python containers to work? Could it be able to clean up to make the test working for single java / py sdk container?

Reply:
So this setup method is (or rather will be) shared by PythonFromJava and JavaFromPython tests.

For external transforms, Transform Service will require Python3.8 and Java11 containers (transformServicePythonContainerSuffix and transformServiceJavaContainerSuffix).

The pipeline SDK, we could run tests with multiple Java and Python versions hence we have to build such containers based on the test ( pythonContainerSuffix and javaContainerSuffix).

For example, currently BT test is run using Python3.8 and 3.11 for pipeline SDK and Java 11 for external SDK.

Reply:
Thanks for the explanation. I see, for a single slang test, there are at least two SDK container is working

Reply:
(though it is generally nice to have these version not hard coded)

Reply:
Thanks. Beam ships with these specific versions (Python 3.8 and Java 11) of expansion service containers. I hope to add documentation on developing expansion service containers with additional language versions for users who need that.